pkg/arvo/sys/arvo.hoon

```diff
@@ -291,14 +291,23 @@
   |=(b=beam =*(s scot `path`[(s %p p.b) q.b (s r.b) s.b]))
 ::
 ++  de-beam
+  ~/  %de-beam
   |=  p=path
   ^-  (unit beam)
   ?.  ?=([@ @ @ *] p)  ~
   ?~  who=(slaw %p i.p)  ~
   ?~  des=?~(i.t.p (some %$) (slaw %tas i.t.p))  ~  :: XX +sym ;~(pose low (easy %$))
-  ?~  ved=(slay i.t.t.p)  ~
-  ?.  ?=([%$ case] u.ved)  ~
-  `(unit beam)`[~ [`ship`u.who `desk`u.des `case`p.u.ved] t.t.t.p]
+  ?~  ved=(de-case i.t.t.p)  ~
+  `(unit beam)`[~ [`ship`u.who `desk`u.des `case`u.ved] t.t.t.p]
```

Comment:
It looks like there are extra casts in this line, it'd be nice to clean it up. I think only `ship` and `desk` should be necessary:

```
`[[`ship`u.who `desk`u.des u.ved] t.t.t.p]
```